packages/builder/src/builder.js

```diff
@@ -110,9 +110,18 @@ export default class Builder {
           /[^a-zA-Z?\d\s:]/g,
           ''
         )
+
+        // TODO: remove deprecated check in Nuxt 3
+        if (p.ssr === false) {
+          p.mode = 'client'
+          consola.warn(`ssr in plugin options has been deprecated, please use mode instead`)
+        } else if (!['client', 'server'].includes(p.mode)) {
+          p.mode = 'all'
```

Comment:
A warning about invalid mode can be useful here. Users may accidentally set things like `clien`, ... that we silently translate it to `all`.

Reply:
`all` is not a options that user need to use, it's just a default condition for both client and server.

`client/server` are the only valid value that should be configured in `mode`, if we want to add warning here, we need to another `else if` for `p.mode === undefined`, if it's ok for you, I can add.

Reply:
True, `all` is not what people should use by default. But invalid values (typos) are silently ignored.

Reply:
OK, I'll add the message.

BTW, this can be also added in eslint-plguin

Reply:
:+1: for that idea

Reply:
Yes. Supporting in ESLint plugin would be awesome.) But we also need to have proper CLI warns too.

Reply:
Warn msg added